### PR TITLE
Cleanup Parser tests

### PR DESCRIPTION
**Pull request by vexx32**
# PR Summary

Code cleanup for `Parser.Tests.ps1`

Removed: 1 test (it was not working as intended and was based on a flawed premise).

## PR Context

Cleanup PR in preparation for another PR.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by PoshChan** [comment]
@vexx32, your last commit had 8 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test will check that a path is correctly interpreted when using '..' and '.'  (line 364)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 190
190:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test will check that the parser can handle a mix of forward slashes and back slashes in the path (line 417)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 197
197:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test checks that the asterisk globs as expected. (line 545)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 204
204:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test checks that we can use a range for globbing: [1-2] (line 557)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 211
211:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that & can be used as a parameter as long as it is quoted. (line 1606)
```powershell
Expected strings to be the same, but they were different.
Expected length: 26
Actual length:   28
Strings differ at index 15.
Expected: '&Get-ChildItem;unset;unset'
But was:  '&Get-ChildItem; unset; unset'
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 681
681:         $result | Should -Be "&Get-ChildItem;unset;unset"
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 8 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test will check that a path is correctly interpreted when using '..' and '.'  (line 364)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\Parser.Tests.ps1: line 190
190:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test will check that the parser can handle a mix of forward slashes and back slashes in the path (line 417)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\Parser.Tests.ps1: line 197
197:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test checks that the asterisk globs as expected. (line 545)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\Parser.Tests.ps1: line 204
204:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test checks that we can use a range for globbing: [1-2] (line 557)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\Parser.Tests.ps1: line 211
211:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that & can be used as a parameter as long as it is quoted. (line 1606)
```powershell
Expected strings to be the same, but they were different.
Expected length: 26
Actual length:   28
Strings differ at index 15.
Expected: '&Get-ChildItem;unset;unset'
But was:  '&Get-ChildItem; unset; unset'
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\Parser.Tests.ps1: line 681
681:         $result | Should -Be "&Get-ChildItem;unset;unset"
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 8 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test will check that a path is correctly interpreted when using '..' and '.'  (line 364)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 190
190:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test will check that the parser can handle a mix of forward slashes and back slashes in the path (line 417)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 197
197:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test checks that the asterisk globs as expected. (line 545)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 204
204:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).This test checks that we can use a range for globbing: [1-2] (line 557)
```powershell
Expected 2, but got 0.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 211
211:         $result.Count | Should -Be 2
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that & can be used as a parameter as long as it is quoted. (line 1606)
```powershell
Expected strings to be the same, but they were different.
Expected length: 26
Actual length:   28
Strings differ at index 15.
Expected: '&Get-ChildItem;unset;unset'
But was:  '&Get-ChildItem; unset; unset'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 681
681:         $result | Should -Be "&Get-ChildItem;unset;unset"
```

**Comment by vexx32** [comment]
C'est la vie. Soon as you try to clean up old code, you miss something. 😄

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 1]
```diff
@@ -299,162 +298,362 @@ foo``u{2195}abc
...
+                        2; break; 3
+                    }
+
+                    4; break; 5
+                }
+            '
+            ExpectedResult = "1", "2", "4"
+        }
+    )
+    It "Use break inside of a loop that is inside another loop. (line 945)" -TestCases $TestCases {
```
Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: We have the same name for some tests that is not good.
Also I see that text in $commands was changed that is not good for cleanup PR.

**Review by iSazonov — COMMENTED** [review]
@vexx32 Thanks for your contribution!

The PR is very large for review. Also you change some tests. Could you please split it? At first we could fix only indents.

File: test/powershell/Language/Parser/Parser.Tests.ps1 (hunk 1 above)
Comment (vexx32) [reply test/powershell/Language/Parser/Parser.Tests.ps1]: The text was altered because the test cases were unreadable. I didn't change their functionality, though. 

The formatting is all that changed apart from removal of semicolons and in a few of them I replaced `while (1)` with `while ($true)` which are functionally identical. It may make more sense to put these in a separate PR, but I feel it's largely just cleanup so I'll leave that decision to you. 🙂

**Comment by vexx32** [comment]
@iSazonov I'll look to split out some of the commits to separate PRs. 👍

**Comment by iSazonov** [comment]
@vexx32 Thanks! Tests is sometimes very sensitive (especially in _parser_ tests where even spaces can be important) and we should not change them without much need.

**Comment by vexx32** [comment]
I don't disagree, but if the parser can't handle spaces we've got some issues anyway 😉 

The parser tests are a bit sensitive, indeed, but I don't think that should mean we have _all_ test cases without spaces. Maybe one or two cases to verify that those modes work, but the rest should be relatively clear as to their intent in my opinion.

**Comment by vexx32** [comment]
@iSazonov first split (just whitespace): #9806

**Comment by PoshChan** [comment]
@vexx32, your last commit had 3 failures in `PowerShell-CI-macos`
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that & can be used as a parameter as long as it is quoted. (line 1606)
```powershell
Expected strings to be the same, but they were different.
Expected length: 28
Actual length:   26
Strings differ at index 15.
Expected: '&Get-ChildItem; unset; unset'
But was:  '&Get-ChildItem;unset;unset'
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 674
674:         $result | Should -Be "&Get-ChildItem; unset; unset"
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Run a command with parameters. (line 1621)
```powershell
Expected strings to be the same, but they were different.
Expected length: 17
Actual length:   15
Strings differ at index 4.
Expected: 'set; unset; unset'
But was:  'set;unset;unset'
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 681
681:         $result | Should -Be "set; unset; unset"
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that an alias is resolved before a function. (line 1657)
```powershell
Expected strings to be the same, but they were different.
Expected length: 19
Actual length:   17
Strings differ at index 6.
Expected: 'unset; unset; unset'
But was:  'unset;unset;unset'
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 698
698:         $result | Should -Be "unset; unset; unset"
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 3 failures in `PowerShell-CI-windows`
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that & can be used as a parameter as long as it is quoted. (line 1606)
```powershell
Expected strings to be the same, but they were different.
Expected length: 28
Actual length:   26
Strings differ at index 15.
Expected: '&Get-ChildItem; unset; unset'
But was:  '&Get-ChildItem;unset;unset'
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\Parser.Tests.ps1: line 674
674:         $result | Should -Be "&Get-ChildItem; unset; unset"
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Run a command with parameters. (line 1621)
```powershell
Expected strings to be the same, but they were different.
Expected length: 17
Actual length:   15
Strings differ at index 4.
Expected: 'set; unset; unset'
But was:  'set;unset;unset'
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\Parser.Tests.ps1: line 681
681:         $result | Should -Be "set; unset; unset"
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that an alias is resolved before a function. (line 1657)
```powershell
Expected strings to be the same, but they were different.
Expected length: 19
Actual length:   17
Strings differ at index 6.
Expected: 'unset; unset; unset'
But was:  'unset;unset;unset'
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\Parser.Tests.ps1: line 698
698:         $result | Should -Be "unset; unset; unset"
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 3 failures in `PowerShell-CI-linux`
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that & can be used as a parameter as long as it is quoted. (line 1606)
```powershell
Expected strings to be the same, but they were different.
Expected length: 28
Actual length:   26
Strings differ at index 15.
Expected: '&Get-ChildItem; unset; unset'
But was:  '&Get-ChildItem;unset;unset'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 674
674:         $result | Should -Be "&Get-ChildItem; unset; unset"
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Run a command with parameters. (line 1621)
```powershell
Expected strings to be the same, but they were different.
Expected length: 17
Actual length:   15
Strings differ at index 4.
Expected: 'set; unset; unset'
But was:  'set;unset;unset'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 681
681:         $result | Should -Be "set; unset; unset"
```
ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs).Test that an alias is resolved before a function. (line 1657)
```powershell
Expected strings to be the same, but they were different.
Expected length: 19
Actual length:   17
Strings differ at index 6.
Expected: 'unset; unset; unset'
But was:  'unset;unset;unset'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/Parser.Tests.ps1: line 698
698:         $result | Should -Be "unset; unset; unset"
```

**Comment by vexx32** [comment]
@iSazonov the diffs look considerably better now. 😄 

Do you think it's worthwhile to further split the PR?

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 2]
```diff
@@ -154,55 +160,55 @@ Describe "ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs)"
...
 
     It "This test checks that we can use a range for globbing: [1-2] (line 557)" {
-        $result = ExecuteCommand "get-childitem $TestDrive/dir1\dir2\testdirfile[1-2].txt"
+        $result = ExecuteCommand "Get-ChildItem $TestDrive/dir1\dir2\testdirfile[1-2].txt"
         $result.Count | Should -Be 2
         $result[0].Name | Should -BeExactly "testdirfile1.txt"
         $result[1].Name | Should -BeExactly "testdirfile2.txt"
     }
 
-    It "This test will check that escaping the $ sigil inside single quotes simply returns the $ character. (line 583)" {
```
Comment (TravisEz13) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: why was this removed?

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 3]
```diff
@@ -214,8 +220,6 @@ Describe "ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs)"
     }
 
     Context "Test Unicode escape sequences." {
-        # These tests require the file to be saved with a BOM.  Unfortunately when this UTF8 file is read by
```
Comment (TravisEz13) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: why was this removed?

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
I have two questions

File: test/powershell/Language/Parser/Parser.Tests.ps1 (hunk 2 above)
Comment (vexx32) [reply test/powershell/Language/Parser/Parser.Tests.ps1]: This was removed because the test does nothing. Look at this:
```
        $result = ExecuteCommand "'`$'"
        $result | Should -BeExactly "`$"
```
That first command causes PowerShell to _already evaluate_ the backtick escape as it's passed into the command. `ExecuteCommand` only ever sees the string `'$'`. Contrary to the expectations of whoever wrote this test, the PowerShell instance being tested will only ever see a single-quoted string containing simply the character `$`.

Now, if we actually assess the test based on what it _says_ it should be doing, we can see it's flatly wrong. Single-quoted strings simply do not process escape characters, that's sort of the point. Very simple test from any PS console:
```
'`$' -ne '$'
```

File: test/powershell/Language/Parser/Parser.Tests.ps1 (hunk 3 above)
Comment (vexx32) [reply test/powershell/Language/Parser/Parser.Tests.ps1]: This comment is no longer accurate. PowerShell no longer relies on BOM to process anything, and all files are processed as UTF8 unless some other encoding is specified with the appropriate cmdlet.

**Comment by TravisEz13** [comment]
@vexx32 	Rebased your branch to make sure we have current test results.

**Comment by TravisEz13** [comment]
@PoshChan 	Please remind me in 1 hour

**Review by TravisEz13 — APPROVED** [review]

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 1 hour ago

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 4]
```diff
@@ -429,19 +628,26 @@ foo``u{2195}abc
     }
 
     It "This test will check that the parser throws a syntax error when a foreach loop is not complete. (line 1238)" {
-        { ExecuteCommand '$count=0;$files = $(get-childitem / -filter *.txt );foreach ($i ;$count' } | Should -Throw -ErrorId "ParseException"
+        { ExecuteCommand '$count=0; $files = $(Get-ChildItem / -filter *.txt ); foreach ($i ; $count' } | Should -Throw -ErrorId "ParseException"
```
Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]:
```suggestion
        { ExecuteCommand '$count=0; $files = $(Get-ChildItem / -Filter *.txt ); foreach ($i ; $count' } | Should -Throw -ErrorId "ParseException"
```

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 5]
```diff
@@ -429,19 +628,26 @@ foo``u{2195}abc
...
     }
 
     It "This test will check that the parser throws a syntax error if the foreach loop is not complete. (line 1248)" {
-        { ExecuteCommand '$count=0;$files = $(get-childitem / -filter *.txt );foreach ($i in ;$count' } | Should -Throw -ErrorId "ParseException"
+        { ExecuteCommand '$count=0; $files = $(Get-ChildItem / -filter *.txt ); foreach ($i in ;$count' } | Should -Throw -ErrorId "ParseException"
```
Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]:
```suggestion
        { ExecuteCommand '$count=0; $files = $(Get-ChildItem / -Filter *.txt ); foreach ($i in ;$count' } | Should -Throw -ErrorId "ParseException"
```

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 6]
```diff
@@ -429,19 +628,26 @@ foo``u{2195}abc
...
     }
 
     It "This will test that the parser throws a syntax error if the foreach loop is missing a closing parentheses. (line 1258)" {
-        { ExecuteCommand '$count=0;$files = $(get-childitem / -filter *.txt );foreach ($i in $files ;$count' } | Should -Throw -ErrorId "ParseException"
+        { ExecuteCommand '$count=0; $files = $(Get-ChildItem / -filter *.txt ); foreach ($i in $files ;$count' } | Should -Throw -ErrorId "ParseException"
```
Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]:
```suggestion
        { ExecuteCommand '$count=0; $files = $(Get-ChildItem / -Filter *.txt ); foreach ($i in $files ;$count' } | Should -Throw -ErrorId "ParseException"
```

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 7]
```diff
@@ -454,31 +660,31 @@ foo``u{2195}abc
...
         $result = ExecuteCommand 'testcmd-parserBVT -Property1 set'
-        $result | Should -Be "set;unset;unset"
+        $result | Should -Be "set; unset; unset"
     }
 
     It "Test that typing a number at the command line will return that number. (line 1630)" {
         $result = ExecuteCommand '3'
         $result | Should -Be "3"
-        $result.gettype() | should -Be ([int])
+        $result.GetType() | Should -Be ([int])
```
Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]:
```suggestion
        $result | Should -BeOfType [int]
```

File: test/powershell/Language/Parser/Parser.Tests.ps1 (hunk 2 above)
Comment (vexx32) [reply test/powershell/Language/Parser/Parser.Tests.ps1]: Then we should test for that explicitly, instead of maintaining a misleading test. 🙂

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@vexx32 Thanks for formatting the tests.
